### PR TITLE
Use Firebase mpp SDK and move Authentication to shared

### PR DESCRIPTION
**Pull request by martinbonnin**
Follow up PR is to move `Authentication` outside of the VM so that it can be shared more easily between screens

### CODE REVIEW

**Comment by joreilly** [comment]
Does this potentially make it easier to use from iOS client?

**Comment by martinbonnin** [comment]
@joreilly in the short term, looks like it broke iOS :( 

```
The /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld command returned non-zero exit code: 1.
output:
ld: framework not found FirebaseAuth
```

In the long term I'm hoping we can share some of the auth logic indeed. I'll revisit shortly

File: androidApp/src/main/java/dev/johnoreilly/confetti/account/SignInView.kt [hunk 1]
```diff
@@ -16,12 +16,15 @@ import androidx.compose.ui.platform.LocalContext
...
+import org.koin.compose.koinInject
 
 
 @Composable
 fun SignInRoute(onBackClick: () -> Unit) {
     Box(
-        modifier = Modifier.fillMaxWidth().fillMaxHeight()
+        modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
```
Comment (yschimke) [inline androidApp/src/main/java/dev/johnoreilly/confetti/account/SignInView.kt:27]:
```suggestion
            .fillMaxSize()
```

**Review by yschimke — APPROVED** [review]

File: androidApp/src/main/java/dev/johnoreilly/confetti/account/SignInView.kt (hunk 1 above)
Comment (martinbonnin) [reply androidApp/src/main/java/dev/johnoreilly/confetti/account/SignInView.kt:27]: TIL, thanks!

**Pull request merged**